sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -92,3 +93,75 @@ case class Mode(
   override protected def withNewChildInternal(newChild: Expression): Expression =
     copy(child = newChild)
 }
+
+/**
+ * Mode in Pandas' fashion. This expression is dedicated only for Pandas API on Spark.
+ */
+case class PandasMode(
```

Comment:
existing `Mode` always ignore NAs, and when there are multi modes, it only return one of them

Reply:
Can we dedup the codes?

Reply:
since only `def merge` and `inputTypes` can be reused, so it maybe not worthwhile